### PR TITLE
Added notification (SMS, WhatsApp, Mail)

### PR DESCRIPTION
**Pull request by codepraycode**
# Title: Implementation of notification (SMS, WhatsApp, Email)

## Overview
This pull request includes the implementation of notification features with their respective endpoints:

### Email endpoint
- POST: /gateway/*/notify/mail -> Send mails
   Data
   ```
   {
       subject: string,
       body: string,
       to: string,  
   }
   ```

`to` field should be email address within the main domain (i.e `@nobox.cloud`). This is because our account hasn't been approved, therefore we can't send `to` a email address that is not in same domain as `nobox.cloud`.

We used [Postmark](https://postmarkapp.com) to implement this feature, I have sent the login credentials to @Akintunde102 so you can obtain the server token and set `MAIL_API_KEY` and `MAIL_FROM` in env. [see environment variables](#environement-variables)

`MAIL_FROM` can default to `noreply@nobox.cloud`.

### SMS endpoint
- POST: /gateway/*/notify/sms -> Send SMS messages
   Data
   ```
   {
       body: string,
       to: string,  
   }
   ```
Used [Twilio](https://console.twilio.com/) service.

### WhatsApp endpoint
- POST: /gateway/*/notify/whatsapp -> Send whatsapp messages
   Data
   ```
   {
       body: string,
       to: string,  
   }
   ```
Used [Twilio](https://console.twilio.com/) service.

### Miscellaneous
- POST: /gateway/*/notify/whatsapp/reply -> Handle whatsapp responses
- GET: /gateway/*/notify/whatsapp/callback -> Twilio status callback

### Environement variables

Obtain and add the following parameters:

```
# Twilio
TWILIO_ACCOUNT_SID=""
TWILIO_AUTH_TOKEN=""
TWILIO_BASE_PHONE_NUMBER=""
TWILIO_WHATSAPP_PHONE_NUMBER=""
TWILIO_WHATSAPP_CODE=""

# Email
MAIL_API_KEY=""
MAIL_FROM="noreply@nobox.cloud"
```

### CODE REVIEW

File: src/modules/app/app.module.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ import { LogTrackerService } from '../track-logs/log-tracker.service';
       ClientFunctionsModule,
       GatewayModule,
       ClientUtilsModule,
-      LogTrackerModule
+      LogTrackerModule,
```
Comment (Akintunde102) [inline src/modules/app/app.module.ts:42]: Why did you add this back? any reason?

File: src/modules/gateway/gateway.service.ts [hunk 2]
```diff
@@ -284,4 +293,123 @@ export class GateWayService {
...
+
+    async replywhatsAppMessage(config: SendMessageConfig) {
+        try {
+        const message: MessageInstance = await twilioClient.messages.create({
+            body: config.body,
+            from: TWILIO_BASE_PHONE_NUMBER,
+            to: config.to,
+        });
+
+        console.debug(message);
```
Comment (Akintunde102) [inline src/modules/gateway/gateway.service.ts:387]: Stop using console, use `this.slog`

File: src/modules/app/app.module.ts (hunk 1 above)
Comment (codepraycode) [reply src/modules/app/app.module.ts:42]: I'll remove that sir
